src/main/scala/translating/ReadELFLoader.scala

```diff
@@ -76,7 +76,7 @@ object ReadELFLoader {
   }
 
   def visitSymbolTableRow(ctx: SymbolTableRowContext): Option[SpecGlobal] = {
-    if (ctx.entrytype.getText == "OBJECT" && ctx.bind.getText == "GLOBAL" && ctx.vis.getText == "DEFAULT") {
+    if ((ctx.entrytype.getText == "OBJECT" || ctx.entrytype.getText == "FUNC") && ctx.bind.getText == "GLOBAL" && ctx.vis.getText == "DEFAULT") {
```

Comment:
We really should be distinguishing between objects and functions in the symbol table - it will confuse things if we just treat them the same like this.

Reply:
Do you have any advice on the best way to do this? I changed SpecGlobal to a trait and extended it with two 
case classes SpecObejct and SpecFunc. But I haven't changed the behaviour of either case class from the trait.

Reply:
I don't think splitting SpecGlobal into two separate case classes really does anything to help, I'd have to look at what it's actually used for in the DSA but that's quite complex.

Reply:
Sorry I assumed you wanted to distinguish between the two for other parts of BASIL not the DSA itself. 
I don't believe we need to distinguish between global objects and functions for DSA. It's convenient to treat them the same since they are accessed similarly (constant base address + constant offset) and we can reuse the globals recognition functionality. 
I can only see two reason we might want to do this:

1. if we extend DSA to perform call resolution it may be useful but even then we can recognise functions by the fact that their corresponding DS node is used in an indirect call
2.  we can also check that object and function DS nodes are not used in wrong manner (e.g. an object  is used in an indirect call), however, I'm not sure how useful this would be

Reply:
Rather than splitting SpecGlobal, it would make more sense to make a new parent trait which contains both SpecGlobal and the new case class for the symbol table function entries. We should probably rename SpecGlobal too since it's very disconnected from its original use now.